lib/utils/fs_windows.go

```diff
@@ -29,5 +31,10 @@ limitations under the License.
 const lockPostfix = ".lock.tmp"
 
 func getPlatformLockFilePath(path string) string {
+	// If target file is itself dedicated lockfile, we don't create another lockfile, since
```

Comment:
Main usage until now was when we locked on some significant file itself (fx `known_hosts`). In Windows, because of how it behaved, we had to add creating additional lockfiles instead of locking on the file itself. Here we have dedicated lockfile, so it seems unnecessary create new lockfile for locking on already dedicated lockfile.

Reply:
Sure, I can see how we can try locking `known_hosts`, but why would we try to lock `known_hosts.lock`?

Reply:
@AntonAm is on vacations but this is a limitation of Windows.
When the lock syscall executes, it won't be possible to write anything to the locked file as it hangs until the lock is released.

Due to this problem, you must lock a different file.